core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigurationSetup.java

```diff
@@ -192,11 +192,16 @@ public ConfigurationBuildItem initializeConfiguration(
     private static <T> void withConverterHelper(final SmallRyeConfigBuilder builder, final Class<T> type, final int priority,
             final Class<? extends Converter<?>> converterClass) {
         try {
-            builder.withConverter(type, priority, ((Class<? extends Converter<T>>) converterClass).newInstance());
+            builder.withConverter(type, priority,
+                    ((Class<? extends Converter<T>>) converterClass).getDeclaredConstructor().newInstance());
         } catch (InstantiationException e) {
             throw toError(e);
         } catch (IllegalAccessException e) {
             throw toError(e);
+        } catch (InvocationTargetException e) {
```

Comment:
This isn't the right handling for this.  The cause of the ITE should be extracted and rethrown, wrapping with `UndeclaredThrowableException` in the event that it's a `Throwable` that isn't a `RuntimeException` or `Error`.

Reply:
OK, see if the new version (force pushed) is better? There is a hiccup though... see the TODO: It's not possible (AFAIK) to re-throw the `InvocationTargetException` as either a `RuntimeException` OR an `Error`, with those `toError`method signatures _returning_ instead of _throwing_. Do you want a PR with that changed? Or just forget about using `ReflectUtil` and inline it for that case,with x2 `throw`?

Reply:
Right it should be throwing the *cause*, like this:

```java
// ...
} catch (InvocationTargetException e) {
    try {
        throw e.getCause();
    } catch (RuntimeException | Error e2) {
        throw e2;
    } catch (Throwable t) {
        throw new UndeclaredThrowableException(t);
    }
}
```

It's wordy but it is the "most" correct.

Reply:
done